packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -1725,6 +1726,45 @@ function validateAttributes(
     }
 }
 
+function validateSlotAttribute(
+    ctx: ParserCtx,
+    parsedAttr: ParsedAttribute,
+    parentNode: ParentNode,
+    element: BaseElement
+): void {
+    const slotAttr = parsedAttr.get('slot');
+
+    if (!slotAttr) {
+        return;
+    }
+
+    function isElementOrSlot(node: ParentNode): node is Element | Slot {
+        return ast.isElement(node) || ast.isSlot(node);
+    }
```

Comment:
Apparently I didn't test yarn build and the rollup typescript plugin doesn't infer the type predicate yet.